scripts/superdesk-archive/module.js

```diff
@@ -540,12 +540,23 @@
                     label: gettext('Spike Item'),
                     icon: 'trash',
                     monitor: true,
-                    controller: ['spike', 'data', '$rootScope', function spikeActivity(spike, data, $rootScope) {
-                        return spike.spike(data.item).then(function(item) {
-                            $rootScope.$broadcast('item:spike');
-                            return item;
-                        });
-                    }],
+                    controller: ['spike', 'data', '$rootScope', 'modal', '$location',
```

Comment:
Please simplify this as:
```js
controller: ['spike', 'data', '$rootScope', 'modal', '$location', '$q',
    function spikeActivity(spike, data, $rootScope, modal, $location, $q) {
        var txt = gettext('Do you want to delete the item permanently? ');
        var isPersonal = $location.path() === '/workspace/personal';
        return $q.when(isPersonal ? modal.confirm(txt) : 0)
            .then(function() {
                return spike.spike(data.item).then(function(item) {
                    $rootScope.$broadcast('item:spike');
                    return item;
                });
            });
    }],
```
It will contain less repetition and will be easier to follow imo.